### PR TITLE
chore: fix typos

### PR DESCRIPTION
**Pull request by adrien-fichet**
Hello, I found some typos in the documentation and initially wanted to fix just a few of them. I then ran `codespell` on the repo and found a bunch more. Please tell me if some files should not be touched.

### CODE REVIEW

**Review by jedevc — DISMISSED** [review]

File: sdk/java/dagger-codegen-maven-plugin/src/main/resources/schemas/schema-0.9.3.json
Comment (jedevc) [inline sdk/java/dagger-codegen-maven-plugin/src/main/resources/schemas/schema-0.9.3.json:1]: Oops, just noticed this now - we shouldn't modify these java schema files - they're for old versions of the API, which would still have these typos.

**Comment by vvaswani** [comment]
The changes in the public docs `docs\current_docs` lgtm. Will defer to others on the changes in code files.

I would also like to have this as a lint check.

**Comment by vikram-dagger** [comment]
Changes in the public docs `docs\current_docs` lgtm. I will defer to others regarding changes in code files. Thanks for the PR @adrien-fichet !

**Review by jedevc — APPROVED** [review]

**Review by helderco — APPROVED** [review]

**Comment by jedevc** [comment]
Going to merge this, since it touches quite a few files, and rebasing this constantly will be a pain :cry:

**Pull request merged**